### ISSUE TITLE
testing: implement test telemetry

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#50414]
Currently, unit test statistics are barely tracked. TeamCity kind of tracks them, but its interface is hard to use programmatically (or at all). GitHub tracks test failures, but it's also hard to parse GitHub to figure out what happened historically.

We should extend our unit test harness on TeamCity to send statistics about test failures, passes, and timeouts to an external database. The database can then be used to analyze our test patterns and determine what needs improvement.

Jira issue: CRDB-4124

**Comment by asubiotto** [comment]
Something I'd really like to see is test runtime over time (including green builds). This would be especially useful for logic tests when determining whether a change caused tests to start timing out, or if the runtime had slowly been growing until hitting a threshold.
